xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.h

```diff
@@ -104,6 +105,7 @@ class CProcessorHD : public ID3DResource
   D3D11_VIDEO_PROCESSOR_RATE_CONVERSION_CAPS m_rateCaps;
   D3D11_TEXTURE2D_DESC m_texDesc;
   PROCESSOR_VIEW_TYPE m_eViewType;
+  std::vector<AVPixelFormat> m_formats;
```

Comment:
m_formats only ever had a single purpose: to tell sw decoder if it needs to convert the output of ffmpeg decoder. a supported format should not depend on any settings. 
nevertheless you can query settings in windows' implementation of CProcessInfo::GetRenderFormats. but if you need that, something goes really wrong.